e2etests/tests/evpn_l2.go

```diff
@@ -90,108 +91,145 @@ var _ = Describe("Routes between bgp and the fabric", Ordered, func() {
-			{exec: hostARedExecutor, from: "hostARed", to: "firstPod", fromIP: infra.HostARedIPv4, toIP: firstPodIP},
+			{exec: podExecutor, from: "firstPod", to: "secondPod", fromIPs: tc.firstPodIPs, toIPs: tc.secondPodIPs},
+			{exec: secondPodExecutor, from: "secondPod", to: "firstPod", fromIPs: tc.secondPodIPs, toIPs: tc.firstPodIPs},
+			{exec: podExecutor, from: "firstPod", to: "hostARed", fromIPs: tc.firstPodIPs, toIPs: tc.hostARedIPs},
+			{exec: podExecutor, from: "firstPod", to: "hostBRed", fromIPs: tc.firstPodIPs, toIPs: tc.hostBRedIPs},
+			{exec: secondPodExecutor, from: "secondPod", to: "hostARed", fromIPs: tc.secondPodIPs, toIPs: tc.hostARedIPs},
+			{exec: secondPodExecutor, from: "secondPod", to: "hostBRed", fromIPs: tc.secondPodIPs, toIPs: tc.hostBRedIPs},
+			{exec: hostARedExecutor, from: "hostARed", to: "firstPod", fromIPs: tc.hostARedIPs, toIPs: tc.firstPodIPs},
 		}
 
 		for _, test := range tests {
 			By(fmt.Sprintf("checking reachability from %s to %s", test.from, test.to))
-			checkPodIsReacheable(test.exec, test.fromIP, test.toIP)
+			Expect(test.fromIPs).To(HaveLen(len(test.toIPs)))
+			for i, fromIP := range test.fromIPs {
+				checkPodIsReacheable(test.exec, fromIP, test.toIPs[i])
+			}
 		}
-	})
+	},
+		Entry("for single stack ipv4", testCase{
+			l2GatewayIPs: []string{"192.171.24.1/24"},
+			firstPodIPs:  []string{"192.171.24.2/24"},
+			secondPodIPs: []string{"192.171.24.3/24"},
+			hostARedIPs:  []string{infra.HostARedIPv4},
+			hostBRedIPs:  []string{infra.HostBRedIPv4},
+		}),
+		Entry("for dual stack", testCase{
+			l2GatewayIPs: []string{"192.171.24.1/24", "fd00:10:245:1::1/64"},
+			firstPodIPs:  []string{"192.171.24.2/24", "fd00:10:245:1::2/64"},
+			secondPodIPs: []string{"192.171.24.3/24", "fd00:10:245:1::3/64"},
+			hostARedIPs:  []string{infra.HostARedIPv4, infra.HostARedIPv6},
+			hostBRedIPs:  []string{infra.HostBRedIPv4, infra.HostBRedIPv6},
+		}),
+		Entry("for single stack ipv6", testCase{
+			l2GatewayIPs: []string{"192.171.24.1/24", "fd00:10:245:1::1/64"},
+			firstPodIPs:  []string{"192.171.24.2/24", "fd00:10:245:1::2/64"},
+			secondPodIPs: []string{"192.171.24.3/24", "fd00:10:245:1::3/64"},
+			hostARedIPs:  []string{infra.HostARedIPv4, infra.HostARedIPv6},
+			hostBRedIPs:  []string{infra.HostBRedIPv4, infra.HostBRedIPv6},
+		}),
+	)
 })
 
-func removeGatewayFromPod(pod *corev1.Pod) {
+func removeGatewayFromPod(pod *corev1.Pod) error {
 	exec := executor.ForPod(pod.Namespace, pod.Name, "agnhost")
-	_, err := exec.Exec("ip", "route", "del", "default", "dev", "eth0")
-	Expect(err).NotTo(HaveOccurred(), "Failed to remove gateway from pod %s", pod.Name)
+	output, err := exec.Exec("ip", "route", "del", "default", "dev", "eth0")
+	Expect(err).NotTo(HaveOccurred(), "Failed to remove ipv4 gateway from pod %s: %s", pod.Name, output)
+	// Extract the ipv6 deafult gw for eth0 nexthop ip since we need it to delete
```

Comment:
can you expand this? Also, is this really required if we are the ones assigning the default gw?